contracts/pools/idle/StableSwapIDLE.vy

```diff
@@ -0,0 +1,998 @@
+# @version 0.2.8
+"""
+@title Curve IDLE Pool
+@author Curve.Fi
+@license Copyright (c) Curve.Fi, 2020 - all rights reserved
+@notice Pool for swapping between Idle Tokens (IdleDAI, IdleUSDC, IdleUSDT)
+"""
+
+# External Contracts
+interface IdleToken:
+    def mintIdleToken(_amount: uint256, _skipWholeRebalance: bool, _referral: address) ->uint256: nonpayable
+    def redeemIdleToken(_amount: uint256) -> uint256: nonpayable
+    def tokenPrice() -> uint256: view
+
+interface CurveToken:
+    def mint(_to: address, _value: uint256) -> bool: nonpayable
+    def burnFrom(_to: address, _value: uint256) -> bool: nonpayable
+
+interface ERC20:
+    def transfer(_to: address, _value: uint256): nonpayable
+    def transferFrom(_from: address, _to: address, _value: uint256): nonpayable
+    def totalSupply() -> uint256: view
+    def balanceOf(_addr: address) -> uint256: view
+
+
+# Events
+event TokenExchange:
+    buyer: indexed(address)
+    sold_id: int128
+    tokens_sold: uint256
+    bought_id: int128
+    tokens_bought: uint256
+
+event TokenExchangeUnderlying:
+    buyer: indexed(address)
+    sold_id: int128
+    tokens_sold: uint256
+    bought_id: int128
+    tokens_bought: uint256
+
+event AddLiquidity:
+    provider: indexed(address)
+    token_amounts: uint256[N_COINS]
+    fees: uint256[N_COINS]
+    invariant: uint256
+    token_supply: uint256
+
+event RemoveLiquidity:
+    provider: indexed(address)
+    token_amounts: uint256[N_COINS]
+    fees: uint256[N_COINS]
+    token_supply: uint256
+
+event RemoveLiquidityOne:
+    provider: indexed(address)
+    token_amount: uint256
+    coin_amount: uint256
```

Comment:
`token_supply`, too?

Reply:
Agreed, this event is lacking a couple of fields.  But it's the same as what we've been using on previous pools so I think this is how the FE expects it to be formatted.